### PR TITLE
add Github Action workflow to run the tests

### PR DESCRIPTION
**Pull request by Opixelum**
- Added yml file for setting up GitHub Actions with smart contract testing
- Refactored smart contract test files with OZ's Prettier config

### CODE REVIEW

**Comment by Opixelum** [comment]
@n1c01a5 @BLOCKCHAINDEVWEB can you review my PR please ?

File: test/feature-erc20.js [hunk 1]
```diff
@@ -52,7 +52,7 @@ beforeEach(async function () {
     receiver6,
     challenger2,
     sender7,
-    receiver7
+    receiver7,
```
Comment (n1c01a5) [inline test/feature-erc20.js:55]: Remove this file of this commit.

**Review by n1c01a5 — CHANGES_REQUESTED** [review]

File: test/feature-erc20.js (hunk 1 above)
Comment (Opixelum) [reply test/feature-erc20.js:55]: I removed the test command for the CI yml file, but you want to cancel the feature-erc20.js formatting ?

Comment (n1c01a5) [reply test/feature-erc20.js:55]: Nop just don't commit this file.

**Comment by n1c01a5** [comment]
Merge with https://github.com/feature-sh/smart-contract/pull/15

**Pull request closed without merging**